### ISSUE TITLE
hub clone prompts for credentials for public repos

### ISSUE THREAD

**Issue by p-mongo** [github/hub#1811]
I tried cloning https://github.com/rspec/rspec-core:

<pre>
% hub clone rspec/rspec-core    
github.com username: a
github.com password for a (never stored): 
Error getting commit patch: Unauthorized (HTTP 401)
Bad credentials
</pre>

Hub should clone public repositories without prompting for credentials.

Hub 2.5.0 installed via its install script.

**Comment by p-mongo** [comment]
The prompt says password is never stored. The most obvious explanation is the application asks for a password every time, which as you might imagine is rather annoying when trying to clone a *public* repo.

If hub indeed needs the password one time to set up some state it could perhaps communicate that.

**Comment by mislav** [comment]
Fair point. Ideas outlined in #225 will improve this messaging.

**Comment by wongww** [comment]
I agree with @p-mongo! I've been trying to figure out what the authentication was for since I was cloning a public repo. Thank you for this clarification; I think it would definitely help new users if this was stated somewhere
